docs/developer_guide/concepts.md

```diff
@@ -159,3 +159,99 @@ After running this playbook, the developer will be invited to manually:
 
 * customize variables and files as necessary,
 * commit everything to the repository.
+
+## Supported object kinds
+
+Arnold applications can declare various kind of objects that will be deployed on the
+openshift cluster.
+
+### DeploymentConfig
+
+[DeploymentConfig](https://docs.openshift.com/container-platform/3.11/dev_guide/deployments/how_deployments_work.html)
+objects are declared in jinja2 template files matching the pattern `dc*.yml.j2` .
+
+> Deployment configurations describe the desired state of a particular component of
+ the application as a pod template.
+
+For the purpose of our blue/green deployment process, the name of a
+`DeploymentConfig` is dynamic: it is suffixed with a stamp that is unique to the version of the deployed application.
+
+### Service (aka `Blue/Green service`)
+
+[Service](https://kubernetes.io/docs/concepts/services-networking/service/)
+objects are declared in jinja2 template files matching the pattern `svc*.yml.j2`
+
+They are part of our blue/green our deployment process and their name are also dynamic: they are suffixed with a deployment stamp.
+Each deployed version of an application has its own set of services.
+
+Services allow components (i.e. `pods`) of a deployed application to communicate with each other while also
+ensuring that the application stack is consistent within the same version.
+
+### Static Service
+
+*Static Service* objects are declared in jinja2 template files matching the pattern `static-svc*.yml.j2`.
+
+Technically, they are pure kubernetes [Service](https://kubernetes.io/docs/concepts/services-networking/service/) objects.
+But in arnold, we use them for a specific case in our blue/green compatible applications.
+
+B/G Services are helpful for blue/green deployments and for internal communication inside the same application, but their dynamic name make it difficult to be used by other applications.
+That's why we introduced static services.
+
+Static services, as their name suggests, are services with a name that does not change.
+
+Static services are declined in 3 variants (i.e. the static service `foo` is declined in `foo-previous`, `foo-current` and `foo-next`).
+Each variant targets the corresponding version of the deployed application.
```

Comment:
sometimes you say version sometimes you say stack. See above comment.

Reply:
I think it's better now, please tell me if it's still confusing.